### PR TITLE
fix(Blocks): remove pop up link and reopening on save/publish

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

- Removing old pop-up with edit/delete options.
- Replacing this pop-up with existing save/cancel pop-up, with an addition of 'Remove' button.
- Focus link URL on opening link pop-up.
- Validation check added on save, so user has to enter a valid URL [reference](https://www.figma.com/file/MWPKHNT98LLWHGRp3MyWqu?node-id=1339:17473&mode=design#643292761)
- Fixed issue of reopening pop-up on save/publish or copy-paste a link, [Issue](https://strapi-inc.atlassian.net/browse/CS-391)

### How to test it?
[Designs](https://strapi-inc.atlassian.net/browse/CS-401)
1. Check by adding a new link by clicking a toolbar link option.
2. Check by copy-paste a link
3. Clicking save/publish a content should not open link pop-up.

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx [hunk 1]
```diff
@@ -51,35 +40,45 @@ const LinkContent = React.forwardRef<HTMLAnchorElement, LinkContentProps>(
...
     const [linkText, setLinkText] = React.useState(elementText);
     const [linkUrl, setLinkUrl] = React.useState(link.url);
+    const linkInputRef = React.useRef<HTMLInputElement>(null);
+    const [showRemoveButton, setShowRemoveButton] = React.useState(false);
+    const [invalidLinkError, setInvalidLinkError] = React.useState('');
 
     const handleOpenEditPopover: React.MouseEventHandler<HTMLAnchorElement> = (e) => {
       e.preventDefault();
       setPopoverOpen(true);
+      setShowRemoveButton(true);
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx:54]: I'm guessing the only case where this won't be true is if you create a new link via the toolbar. But I can't find how the popover would be opened in that case. Could you explain?

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx [hunk 2]
```diff
@@ -51,35 +40,45 @@ const LinkContent = React.forwardRef<HTMLAnchorElement, LinkContentProps>(
...
+        editor.lastInsertedLinkPath = null;
+      } catch (error) {
+        e.preventDefault();
+
+        setInvalidLinkError(
+          formatMessage({
+            id: 'components.Blocks.popover.link.error',
+            defaultMessage: 'Please enter valid link',
+          })
+        );
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx:81]: The UX chosen in the designs is different, it's to just disable the field if the value is invalid and not show an error message

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx (hunk 1 above)
Comment (madhurisandbhor) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx:53]: Its the same popover we show on clicking new link via toolbar. So on clicking we are inserting a new link and save its path, if current path is same as last inserted link path we show the popover. @Feranchz can explain more if its not clear.

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx [hunk 3]
```diff
@@ -46,40 +45,43 @@ const LinkContent = React.forwardRef<HTMLAnchorElement, LinkContentProps>(
...
       setShowRemoveButton(true);
     };
 
-    const handleSave: React.FormEventHandler = (e) => {
-      e.stopPropagation();
+    const onLinkChange: React.FormEventHandler = (e) => {
+      setIsSaveDisabled(false);
+      setLinkUrl(e.target.value);
+
+      const delayValidation = setTimeout(() => {
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx:60]: Why do we need the timeout? Aren't we able to run that logic directly on every update?

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx [hunk 4]
```diff
@@ -46,40 +45,43 @@ const LinkContent = React.forwardRef<HTMLAnchorElement, LinkContentProps>(
...
+        } catch (error) {
+          e.preventDefault();
+          setIsSaveDisabled(true);
+        }
+      }, 500);
 
-      try {
-        // eslint-disable-next-line no-new
-        new URL(linkUrl);
+      return () => clearTimeout(delayValidation);
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx:70]: I don't think this returned function is ever called

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx (hunk 3 above)
Comment (madhurisandbhor) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx:60]: ohh you're right!

Comment (madhurisandbhor) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx:60]: I think this type of validation is apt for search and so I tend to think and apply same logic 😒

**Review by remidej — APPROVED** [review]

**Pull request merged**